tests/tests_simple_settings.yml

```diff
@@ -1,31 +1,50 @@
 ---
-- name: set parameters for test
+- name: test simple kernel settings
   hosts: all
   tasks:
-    - name: set vars used by this test
+    - name: set platform independent vars used by this test
       include_vars:
         file: vars/vars_simple_settings.yml
 
-- name: apply the settings
-  hosts: all
-  roles:
-    - linux-system-roles.kernel_settings
+    - name: set platform specific vars used by this test
+      include_vars:
+        file: "{{ item }}"
+      loop: "{{
+        q('first_found', __kernel_settings_test_vars_files, errors='ignore')
+      }}"
+      vars:
+        __kernel_settings_test_vars_files:
+          - "vars/vars_simple_settings_{{ ansible_distribution }}_\
+            {{ ansible_distribution_major_version }}.yml"
+          - "vars/vars_simple_settings_{{ ansible_distribution }}.yml"
```

Comment:
I feel this is overcomplicated for the task (skipping a test on Fedora 31). A single `when` condition on `include_role` should be enough, like here: https://github.com/linux-system-roles/network/blob/master/tests/tests_states_nm.yml#L20 . (Fedora 32 works fine, by the way?)

Reply:
I don't want to skip the entire test on Fedora (note that there is a bootloader cmdline problem on both Fedora 31 and Fedora 32 cloud images) - I want to test everything *except* the bootloader cmdline settings on Fedora - I suppose I could revert back to having all of the test settings in the single `vars/vars_simple_settings.yml`, and do something like this?
```yaml
- name: disable bootloader cmdline testing on Fedora
  set_fact:
    kernel_settings_bootloader_cmdline: none
    __kernel_settings_blcmdline_value: none
    __kernel_settings_check_reboot: false
  when: ansible_distribution == "Fedora"
```
?

Reply:
Sorry, I did not understand that the error persists on Fedora 32 as well - is it fixed in rawhide, or what is your strategy for supporting Fedora? The problem should not remain unfixed forever.

Reply:
> Sorry, I did not understand that the error persists on Fedora 32 as well - is it fixed in rawhide, or what is your strategy for supporting Fedora? The problem should not remain unfixed forever.

I will have a follow-on PR which is for the documentation, explaining that the Fedora Cloud images for F31 and F32 are problematic:
* Fedora uses BootLoaderSpec config snippets in /boot/loader/config/, but the initial /boot/loader/config/$machineid-$version.conf file machineid does not match the /etc/machine-id in the image.  This causes the kernel_settings role (via tuned) not to set the bootloader cmdline properly.  This is fixed by doing a `dnf update` of the `grub*` packages.
* Fedora builds the cloud images with a buggy bootloader that does not add a space after user settable bootloader variables.  This causes the kernel_settings role (via tuned) not to set the bootloader cmdline properly.  This is fixed by doing a `dnf update` of the `grub*` packages and rebooting.

My hope is that
* Fedora will rebuild and publish newer cloud images - I am trying to find out where to request that to be done
* Fedora 33 will finally fix the bootloader in the cloud images

Reply:
@richm I am especially interested in the last point - is rawhide fixed already? If not, what fixes does it need?

Reply:
> @richm I am especially interested in the last point - is rawhide fixed already? If not, what fixes does it need?

I tried http://mirrors.kernel.org/fedora/development/rawhide/Cloud/x86_64/images/Fedora-Cloud-Base-Rawhide-20200603.n.0.x86_64.qcow2 - this is even more broken than the f31 and f32 images - none of the workarounds help.